xapian-core/cluster/cluster.cc

```diff
@@ -110,7 +114,7 @@ TermListGroup::add_document(const Document &document) {
 }
 
 void
-TermListGroup::add_documents(DocumentSource &docs) {
+TermListGroup::add_documents(MSetDocumentSource docs) {
```

Comment:
The point of using `DocumentSource` was to allow using this without driving it from an MSet. If you think that's too much abstraction for the moment, then get rid of the abstraction completely by passing an MSet directly. Otherwise, you want to keep the abstract base class as the way you're passing things around.